File: frontend/src/types/outreach.ts
export type Outreach = {
  id: number
  to_email: string
  subject: string
  body: string
  status: string
  created_at: string
  updated_at: string
}

export type OutreachCreate = {
  to_email: string
  subject: string
  body: string
}

File: backend/app/schemas/outreach.py
from datetime import datetime
from pydantic import BaseModel, EmailStr, ConfigDict


class OutreachBase(BaseModel):
    to_email: EmailStr
    subject: str
    body: str
    status: str = "queued"


class OutreachCreate(OutreachBase):
    pass


class Outreach(OutreachBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


